packages/mgt-components/src/components/mgt-tasks/mgt-tasks.ts

```diff
@@ -984,88 +970,85 @@ export class MgtTasks extends MgtTemplatedComponent {
     if (folders.length > 0 && !this._newTaskFolderId) {
       this._newTaskFolderId = folders[0].id;
     }
+
+    const folderOptions = html`
+      ${repeat(
+        folders,
+        folder => folder.id,
+        folder => html`<fluent-option value="${folder.id}">${folder.name}</fluent-option>`
+      )}`;
+
     const taskFolder = this._currentFolder
       ? html`
           <span class="NewTaskBucket">
-            ${this.renderBucketIcon()}
+            ${this.renderBucketIcon(iconColor)}
             <span>${this.getFolderName(this._currentFolder)}</span>
           </span>
         `
       : html`
-          <span class="NewTaskBucket">
-            ${this.renderBucketIcon()}
-            <select aria-label="new task bucket"
-              .value="${this._newTaskFolderId}"
-              @change="${(e: Event) => {
-                this._newTaskFolderId = (e.target as HTMLInputElement).value;
-              }}"
-            >
-              ${folders.map(
-                folder => html`
-                  <option value="${folder.id}">${folder.name}</option>
-                `
-              )}
-            </select>
-          </span>
-        `;
+         <fluent-select>
+          <span slot="start">${this.renderBucketIcon(iconColor)}</span>
+          ${folders.length > 0 ? folderOptions : html`<fluent-option selected>No folders found</fluent-option>`}
+        </fluent-select>`;
+
+    const handleDateChange = (e: UIEvent) => {
```

Comment:
Why don't we make this a class method as avoid re-creating it on each render?

Reply:
I'm a little curious about this. I can't find the right material to read more on. My understanding at the moment is on rendering, functions which appear to be invoked on attributes/properties are anonymously _created_. For instance:

```html
// this has an anonymous function created on render
<div @click={ (e) => someFunction(e)}

// this doesn't
<div @click={someFunction}
```
So is the same thing happening when you declare an inline function? Or what does re-created here mean? 

Also, in some classes I do have this scenario:

```typescript,html
function funcClick(e: Event){
  console.log(" this is an ouside function called successfully");
}

class MgtTask extends MgtComponent {
  // takes in a function arg from somewhere else that executes with the event
  private handleClick(e: Event, fn: SomeClickFunctionType){
    e.preventDefault();
    fn(e)
    // some more functionality
  }

  render(){
    return html`
      <div @click=${(e: Event) => this.handleClick(e, funcClick} // <- Is this anonymously re-created?
    `
  }
}
```